e2e-tests/src/validators.rs

```diff
@@ -138,9 +138,20 @@ pub async fn prepare_validators<S: SignedConnectionApi + AuthorRpc>(
         }));
     }
 
-    for controller in accounts.controller_raw_keys.iter() {
+    let address_tokens = node.split(':').collect::<Vec<_>>();
+    let ws = address_tokens[0];
+    let ip_address = address_tokens[1];
```

Comment:
Oh gods, I just noticed this totally isn't an IP address, but something sinister. Dx Maybe `let address_prefix = format!("{}:{}", address_tokens[0], address_tokens[1])` to avoid confusion? You only modify the port anyway.